ee/allowedcmd/cmd.go

```diff
@@ -13,17 +13,55 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+
+	"github.com/kolide/launcher/pkg/traces"
 )
 
-type AllowedCommand func(ctx context.Context, arg ...string) (*exec.Cmd, error)
+type AllowedCommand func(ctx context.Context, arg ...string) (*TracedCmd, error)
+
+type TracedCmd struct {
+	*exec.Cmd
```

Comment:
one flaw here is that someone could just say `TracedCmd.Cmd.Run()` and bypass the func override, considered making it a private var, but there are a lot of places where we do need to access the embedded `*exec.Cmd` to add Env, Std in/out/err pipes, etc and it felt messy to try and implement all those funcs ... but make worth it?